sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -659,14 +669,7 @@ class ResolveSessionCatalog(
   object SessionCatalogAndTable {
     def unapply(nameParts: Seq[String]): Option[(CatalogPlugin, Seq[String])] = nameParts match {
       case SessionCatalogAndIdentifier(catalog, ident) =>
-        if (nameParts.length == 1) {
```

Comment:
so we are going to remove the hack in `TempViewOrV1Table` in a followup PR?

Reply:
OK. I will do it as a follow up. (I couldn't just remove the check because of `SPARK-30799: temp view name can't contain catalog name`, but I will think about it as a follow up).